### PR TITLE
feat(kueue): add logic to enable new component

### PR DESCRIPTION
**Pull request by zdtsw**
will need https://github.com/opendatahub-io/opendatahub-operator/pull/826 in together
including changes from https://github.com/opendatahub-io/opendatahub-operator/pull/820 

cherry-pick https://github.com/opendatahub-io/opendatahub-operator/pull/826

re-generated manifest to include rbac from featuretracker in PR 801

ref: https://issues.redhat.com/browse/RHOAIENG-2104 

## How Has This Been Tested?
new: with latest code live build quay.io/wenzhou/opendatahub-operator-catalog:v2.7.827-3 is working

old:
live-build quay.io/wenzhou/opendatahub-operator-catalog:v2.7.827-2
Screenshot from 2024-01-29 14-54-09
from operator point of view, no problem to deploy kueue

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: config/manifests/description-patch.yml [hunk 1]
```diff
@@ -16,5 +16,5 @@ spec:
     * Data Science Pipelines - Pipeline solution for end to end MLOps workflows that support the Kubeflow Pipelines SDK and Tekton
     * Model Mesh - ModelMesh Serving is the Controller for managing ModelMesh, a general-purpose model serving management/routing layer
     * Distributed Workloads(Incubation) - Stack built to make managing distributed compute infrastructure in the cloud easy and intuitive for Data Scientists. This stack consists of two components
```
Comment (astefanutti) [inline config/manifests/description-patch.yml:18]: nit: "two" -> "three"

File: config/rbac/role.yaml [hunk 2]
```diff
@@ -1487,3 +1500,63 @@ rules:
   - patch
   - update
   - watch
+- apiGroups:
```
Comment (astefanutti) [inline config/rbac/role.yaml:1503]: I'm not sure, is this really needed? Why the ODH operator would need those permissions?

**Review by astefanutti — APPROVED** [review]

**Review by sutaakar — APPROVED** [review]

File: controllers/dscinitialization/dscinitialization_controller.go.bk [hunk 3]
```diff
@@ -0,0 +1,400 @@
+/*
+Copyright 2023.
```
Comment (astefanutti) [inline controllers/dscinitialization/dscinitialization_controller.go.bk:2]: Is that file a backup that should not be checked in?

File: config/rbac/role.yaml [hunk 4]
```diff
@@ -669,6 +673,15 @@ rules:
...
```
Comment (astefanutti) [inline config/rbac/role.yaml:676]: Is there a corresponding Kubebuilder tag somewhere?

File: config/rbac/role.yaml (hunk 2 above)
Comment (zdtsw) [reply config/rbac/role.yaml]: do not see error after removal

File: controllers/dscinitialization/dscinitialization_controller.go.bk (hunk 3 above)
Comment (zdtsw) [reply controllers/dscinitialization/dscinitialization_controller.go.bk:2]: sorry, mistaken, will remove

File: config/rbac/role.yaml (hunk 4 above)
Comment (zdtsw) [reply config/rbac/role.yaml:676]: for features.opendatahub.io? 
https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/controllers/dscinitialization/dscinitialization_controller.go#L71

Comment (astefanutti) [reply config/rbac/role.yaml:676]: Yes. I see it's likely a leftover from #801.

**Review by astefanutti — APPROVED** [review]
LGTM

**Comment by zdtsw** [comment]
/onhold

waiting for https://github.com/opendatahub-io/kueue/pull/3

**Comment by VaishnaviHire** [comment]
Does Kueue manifests and other DW manifests share any resources? Asking to ensure we handle cases for shared resources.

**Comment by astefanutti** [comment]
> @astefanutti Does Kueue manifests and other DW manifests share any resources? Asking to ensure we handle cases for shared resources.

No, The Kueue component does not share any resources with the CodeFlare component, nor the KubeRay one.

**Review by VaishnaviHire — APPROVED** [review]
/lgtm

**Pull request merged**